filesystem/apps/zfs/vdev_label.rs

```diff
@@ -0,0 +1,1011 @@
+use std::mem;
+
+use super::vdev::VdevLabel;
+
+// vdev_dirty() flags
+const VDD_METASLAB: u64 = 0x01;
+const VDD_DTL: u64 = 0x02;
+
+// Offset of embedded boot loader region on each label
+const VDEV_BOOT_OFFSET: usize = 2 * mem::size_of::<VdevLabel>();
+// Size of embedded boot loader region on each label.
+// The total size of the first two labels plus the boot area is 4MB.
+const VDEV_BOOT_SIZE: usize = 7 << 19; // 3.5M
+
+// Size of label regions at the start and end of each leaf device.
+const VDEV_LABEL_START_SIZE: usize = (2 * mem::size_of::<VdevLabel>() + VDEV_BOOT_SIZE);
+const VDEV_LABEL_END_SIZE: usize = (2 * mem::size_of::<VdevLabel>());
+const VDEV_LABELS: u8 = 4;
+const VDEV_BEST_LABEL: u8 = VDEV_LABELS;
+
+// Basic routines to read and write from a vdev label.
+// Used throughout the rest of this file.
+vdev_label_offset(psize: u64, l: u8, offset: u64) -> u64 {
```

Comment:
Whoops, that was not supposed to get pushed. That file is no longer necessary. There might be other unnecessary files too. I'll submit another PR to remove them.